### PR TITLE
Replace text() function

### PR DESCRIPTION
**Pull request by shweta83**
Replace `text()` function with `normalize-space(.)` to standardize use in xpath locators.
 Issue  https://github.com/SatelliteQE/airgun/issues/591

### CODE REVIEW

**Comment by mshriver** [comment]
@shweta83 how much of the robottelo UI test folder have you executed against this? We can set up a Jenkins job to make it easier since this could be very slow for you to run locally.

The element selection of the XPATH expression becomes a bit more important when using normalize-space(.) as it will gather and concatenate all text from child nodes of the contextual element being selected. `text()` will collect only the text from the selected element.

So its possible the use of `normalize-space(.)` will grab more text than was being collected before, causing test behavior/validation to fail.  These can be really hard to spot on visual inspection/review.  Easy to fix once we find them by adjusting the XPATH to select a more specific element or by using `normalize-space(text())`

File: airgun/views/discoveredhosts.py [hunk 1]
```diff
@@ -147,42 +147,47 @@ def is_displayed(self):
...
+    )
 
 
 class DiscoveredHostsAssignOrganizationDialog(DiscoveredHostsActionDialog):
     """Discovered hosts Assign Organization action dialog view"""
 
     title = Text(
-        "//h4[text()='Assign Organization - The following hosts are about to be changed']"
+        "//h4[normalize-space(.)='Assign Organization -"
+        "The following hosts are about to be changed']"
```
Comment (mshriver) [inline airgun/views/discoveredhosts.py:160]: Looks like this is missing a space between `-` and `The`.

File: airgun/views/hardware_model.py [hunk 2]
```diff
@@ -10,13 +10,13 @@
 
 
 class DeleteHardwareModelDialog(ConfirmationDialog):
-    confirm_dialog = Text(".//button[text()='Delete']")
-    cancel_dialog = Text(".//button[text()='Cancel']")
+    confirm_dialog = Text(".//button[normalize-space(.)='Delete']")
+    cancel_dialog = Text(".//button[normalize-space(.)/='Cancel']")
```
Comment (mshriver) [inline airgun/views/hardware_model.py:14]: Extra slash here?

File: airgun/views/host.py [hunk 3]
```diff
@@ -754,42 +759,49 @@ def is_displayed(self):
 
 class HostsAssignOrganization(HostsActionCommonDialog):
     title = Text(
-        "//h4[text()='Assign Organization - The following hosts are about to be changed']"
+        "//h4[normalize-space(.)='Assign Organization -"
+        "The following hosts are about to be changed']"
```
Comment (mshriver) [inline airgun/views/host.py:763]: Missing space

File: airgun/views/hardware_model.py (hunk 2 above)
Comment (shweta83) [reply airgun/views/hardware_model.py]: Added by mistake. Removed the slash

**Comment by shweta83** [comment]
> @shweta83 how much of the robottelo UI test folder have you executed against this? We can set up a Jenkins job to make it easier since this could be very slow for you to run locally.
> 
> The element selection of the XPATH expression becomes a bit more important when using normalize-space(.) as it will gather and concatenate all text from child nodes of the contextual element being selected. `text()` will collect only the text from the selected element.
> 
> So its possible the use of `normalize-space(.)` will grab more text than was being collected before, causing test behavior/validation to fail. These can be really hard to spot on visual inspection/review. Easy to fix once we find them by adjusting the XPATH to select a more specific element or by using `normalize-space(text())`

@mshriver I totally agree. Though I tested the changes for most of the files but it would be easier if we have Jenkins job to test the changes so that we can fix it if any validation failure occurs.

**Review by mshriver — APPROVED** [review]

**Pull request merged**